android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiToolbar.java

```diff
@@ -120,14 +121,16 @@ public void setToolbarExtendBackground() {
 	 * Handler for extending the background.
 	 */
 	private void handleBackgroundExtended() {
-		Window window = TiApplication.getAppCurrentActivity().getWindow();
-		//Calculate Status bar's height
-		int statusBarHeight = calculateStatusBarHeight();
-		//Add padding to extend the toolbar's background
-		toolbar.setPadding(toolbar.getPaddingLeft(),statusBarHeight + toolbar.getPaddingTop(), toolbar.getPaddingRight(),toolbar.getPaddingBottom());
-		//Set flags for the current window that allow drawing behind status bar
-		window.setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);
-
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+			Window window = TiApplication.getAppCurrentActivity().getWindow();
+			//Calculate Status bar's height
+			int statusBarHeight = calculateStatusBarHeight();
+			//Add padding to extend the toolbar's background
+			toolbar.setPadding(toolbar.getPaddingLeft(),statusBarHeight + toolbar.getPaddingTop(), toolbar.getPaddingRight(),toolbar.getPaddingBottom());
+			//Set flags for the current window that allow drawing behind status bar
+			window.getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+			window.setStatusBarColor(Color.TRANSPARENT);
+		}
```

Comment:
We should remove the `setPadding()` call and instead have the OS automatically adjust for the top statusbar via a call to `View.setFitsSystemWidnows(true)` like this...

```java
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
	// Set flags for the current window that allow drawing behind status bar
	toolbar.setFitsSystemWindows(true);
	Window window = TiApplication.getAppCurrentActivity().getWindow();
	window.getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
	window.setStatusBarColor(Color.TRANSPARENT);
}
```

I've confirmed that the above works on a Motorola "Moto G" (Android 5.0) while the TextField top docked clipboard is shown. See the following screenshots for proof. I'm using a purple toolbar background instead of an image for this test:
- Without TextField clipboard:  [Screenshot](https://axwaysoftware-my.sharepoint.com/personal/jquick_axway_com/_layouts/15/guestaccess.aspx?guestaccesstoken=Raqw73ddGLp5lFce0fKvZ91AXWJusmh%2fyladZHasv4k%3d&docid=2_13834d04a1eab468da630fd51d079a7a1&rev=1)
- With TextField clipboard shown:  [Screenshot](https://axwaysoftware-my.sharepoint.com/personal/jquick_axway_com/_layouts/15/guestaccess.aspx?guestaccesstoken=zGfvxs6l8GpWC7t4HelbzOTsHDPm%2bDQ9Aj6o7Yl3Mmc%3d&docid=2_1da83274cedd34b4e97daa4788ac555b0&rev=1)

I've tested the above code on a Pixel XL (Android 7.1) device as well. The toolbar `extendBackground` correctly covers the top statusbar area on it too.